src/components/Settings/SettingsAddressbookShare.vue

```diff
@@ -22,50 +22,46 @@
 
 <template>
 	<div class="addressbook__shares">
-		<div class="dropdown-menu">
-			<label class="typo__label" for="ajax">Async multiselect</label>
-			<multiselect
-				id="ajax"
-				v-model="selectedUserOrGroup"
-				:options="usersOrGroups"
-				:multiple="true"
-				:searchable="true"
-				:loading="isLoading"
-				:internal-search="false"
-				:clear-on-select="false"
-				:close-on-select="false"
-				:options-limit="250"
-				:limit="3"
-				:limit-text="limitText"
-				:max-height="600"
-				:show-no-results="false"
-				:hide-selected="true"
-				label="name"
-				track-by="code"
-				placeholder="Type to search"
-				open-direction="bottom"
-				@search-change="asyncFind">
-				<template slot="clear" slot-scope="props">
-					<div v-if="selectedUserOrGroup.length" class="multiselect__clear" @mousedown.prevent.stop="clearAll(props.search)" />
-				</template>
-				<span slot="noResult">Oops! No elements found. Consider changing the search query.</span>
-			</multiselect>
-			<pre class="language-json"><code>{{ selectedUserOrGroup }}</code></pre>
-		</div>
+		<multiselect
+			id="users-groups-search"
+			:options="usersOrGroups"
+			:searchable="true"
+			:loading="isLoading"
+			:internal-search="false"
+			:options-limit="250"
+			:limit="3"
+			:max-height="600"
+			:show-no-results="false"
+			:placeholder="placeholder"
+			track-by="match"
+			label="match"
+			open-direction="bottom"
+			class="multiselect-vue"
+			@search-change="asyncFind"
+			@input="shareAddressbook">
+			<template slot="singleLabel" slot-scope="props"><span class="option__desc"><span class="option__title">{{ props.option.matchpattern }}</span></span></template>
+			<template slot="option" slot-scope="props">
+				<div class="option__desc">
+					<span>{{ props.option.matchstart }}</span><span class="shareematch--bold">{{ props.option.matchpattern }}</span><span>{{ props.option.matchend }} {{ props.option.matchtag }}</span>
+				</div>
+			</template>
+			<span slot="noResult">{{ noResult }} </span>
+		</multiselect>
 		<!-- list of user or groups addressbook is shared with -->
 		<ul v-if="addressbook.shares.length > 0" class="addressbook__shares__list">
-			<address-book-sharee v-for="sharee in addressbook.shares" :key="sharee.name" :sharee="sharee" />
+			<address-book-sharee v-for="sharee in addressbook.shares" :key="sharee.displayname + sharee.group" :sharee="sharee" />
```

Comment:
You can also use the index :)
`v-for="(sharee, index) in addressbook.shares" :key="index"`

Both are fine as long as thy're unique. But we can't make sure displayNames are uniques, id are though :)
Pick one ;)